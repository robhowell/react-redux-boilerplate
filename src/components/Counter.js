import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  increaseCounter,
  decreaseCounter,
  startIncreaseCounter,
  startDecreaseCounter
} from '../state/counter/actions';

const Counter = ({
  counter,
  increaseCounter,
  decreaseCounter,
  startIncreaseCounter,
  startDecreaseCounter
}) => (
  <div>
    <p>Counter: {counter}</p>
    <button
      type="button"
      onClick={increaseCounter}
      data-selector="increase-counter-button"
    >
      +1
    </button>
    <button
      type="button"
      onClick={decreaseCounter}
      data-selector="decrease-counter-button"
    >
      -1
    </button>
    <button
      type="button"
      onClick={() => startIncreaseCounter()}
      data-selector="increase-counter-delayed-button"
    >
      +1 (delayed 1s)
    </button>
    <button
      type="button"
      onClick={() => startDecreaseCounter()}
      data-selector="decrease-counter-delayed-button"
    >
      -1 (delayed 1s)
    </button>
  </div>
);

Counter.propTypes = {
  counter: PropTypes.number.isRequired,
  increaseCounter: PropTypes.func.isRequired,
  decreaseCounter: PropTypes.func.isRequired,
  startIncreaseCounter: PropTypes.func.isRequired,
  startDecreaseCounter: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  counter: state.counter
});

const mapDispatchToProps = {
  increaseCounter,
  decreaseCounter,
  startIncreaseCounter,
  startDecreaseCounter
};

export { Counter };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Counter);
