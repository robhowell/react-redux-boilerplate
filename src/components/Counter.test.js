import React from 'react';
import { shallow } from 'enzyme';
import { Counter } from './Counter';

describe('Components/Counter Component', () => {
  const mockProps = {
    counter: 1,
    increaseCounter: jest.fn(),
    decreaseCounter: jest.fn(),
    startIncreaseCounter: jest.fn(),
    startDecreaseCounter: jest.fn()
  };

  const wrapper = shallow(<Counter {...mockProps} />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call increaseCounter when clicking the increase counter button', () => {
    wrapper.find('[data-selector="increase-counter-button"]').simulate('click');
    expect(mockProps.increaseCounter).toHaveBeenCalled();
  });

  it('should call decreaseCounter when clicking the decrease counter button', () => {
    wrapper.find('[data-selector="decrease-counter-button"]').simulate('click');
    expect(mockProps.decreaseCounter).toHaveBeenCalled();
  });

  it('should call startIncreaseCounter when clicking the increase counter delayed button', () => {
    wrapper
      .find('[data-selector="increase-counter-delayed-button"]')
      .simulate('click');
    expect(mockProps.startIncreaseCounter).toHaveBeenCalled();
  });

  it('should call startDecreaseCounter when clicking the increase counter button', () => {
    wrapper
      .find('[data-selector="decrease-counter-delayed-button"]')
      .simulate('click');
    expect(mockProps.startDecreaseCounter).toHaveBeenCalled();
  });
});
