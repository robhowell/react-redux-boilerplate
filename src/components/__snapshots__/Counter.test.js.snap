// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Counter Component should render correctly 1`] = `
<div>
  <p>
    Counter: 
    1
  </p>
  <button
    data-selector="increase-counter-button"
    onClick={[MockFunction]}
    type="button"
  >
    +1
  </button>
  <button
    data-selector="decrease-counter-button"
    onClick={[MockFunction]}
    type="button"
  >
    -1
  </button>
  <button
    data-selector="increase-counter-delayed-button"
    onClick={[Function]}
    type="button"
  >
    +1 (delayed 1s)
  </button>
  <button
    data-selector="decrease-counter-delayed-button"
    onClick={[Function]}
    type="button"
  >
    -1 (delayed 1s)
  </button>
</div>
`;
